src/plugins/data/common/datasets/types.ts

```diff
@@ -88,12 +110,69 @@ export interface DataStructure {
   id: string;
   /** Human-readable name of the data structure */
   title: string;
-  /** The type of the data structure, registered by other classes */
+  /** The type of the data structure */
   type: string;
   /** Optional reference to the parent data structure */
   parent?: DataStructure;
   /** Optional array of child data structures */
   children?: DataStructure[];
+  /** Optional metadata for the data structure */
+  meta?: DataStructureMeta;
+}
+
+/**
+ * Metadata for a data structure, used for additional properties like icons or tooltips.
+ */
+export interface DataStructureMeta {
+  type: DATA_STRUCTURE_META_TYPES;
+}
+
+/**
+ * DataStructureMeta types
+ */
+export enum DATA_STRUCTURE_META_TYPES {
+  FEATURE = 'FEATURE',
+  CUSTOM = 'CUSTOM',
+}
+
+/**
+ * Metadata for a data structure, used for additional properties like icons or tooltips.
+ */
+export interface DataStructureFeatureMeta extends DataSourceMeta {
```

Comment:
```suggestion
export interface DataStructureFeatureMeta extends DataStructureMeta {
```

Reply:
Good catch.

I made the same mistake while making the diagrams and fixed it and just to do the same mistake in the interface